packages/catalyst/src/data/types.ts

```diff
@@ -0,0 +1,88 @@
+import { Filter } from "mongodb";
+import {
+  CatalystConfig,
+  CatalystReferenceField,
+  CatalystRichTextField,
+  CatalystTextField,
+} from "../types";
+
+export type CatalystDataObject<C extends CatalystConfig> =
+  CatalystCollectionDataObject<C> & CatalystGlobalsDataObject<C>;
+
+export type CatalystCollectionDataObject<C extends CatalystConfig> = {
+  [K in keyof C["collections"]]: {
+    findAsUser: CatalystFindDataFunction<C, K>;
+    findOneAsUser: CatalystFindOneDataFunction<C, K>;
+    find: CatalystFindDataFunction<C, K>;
+    findOne: CatalystFindOneDataFunction<C, K>;
+  };
+};
+
+export type CatalystGlobalsDataObject<C extends CatalystConfig> = {
+  [K in keyof C["globals"]]: {
+    getAsUser: CatalystGetDataFunction<C, K>;
+    get: CatalystGetDataFunction<C, K>;
+  };
+};
+
+export type CatalystFindDataFunction<
+  C extends CatalystConfig,
+  K extends keyof C["collections"]
+> = (
+  options?: QueryOptions<
+    ComputedCatalystFields<C, C["collections"][K]["fields"]>
+  >
+) => Promise<
+  (ComputedCatalystFields<C, C["collections"][K]["fields"]> & { _id: string })[]
+>;
+
+export type CatalystFindOneDataFunction<
+  C extends CatalystConfig,
+  K extends keyof C["collections"]
+> = (
+  options?: Omit<
+    WithRequired<
+      QueryOptions<ComputedCatalystFields<C, C["collections"][K]["fields"]>>,
+      "filters"
+    >,
+    "limit"
+  >
+) => Promise<
+  ComputedCatalystFields<C, C["collections"][K]["fields"]> & { _id: string }
+>;
+
+export type CatalystGetDataFunction<
+  C extends CatalystConfig,
+  K extends keyof C["globals"]
+> = (
+  options?: Omit<
+    QueryOptions<ComputedCatalystFields<C, C["globals"][K]["fields"]>>,
+    "limit"
+  >
+) => Promise<ComputedCatalystFields<C, C["globals"][K]["fields"]>>;
+
+type ComputedCatalystFields<
```

Comment:
ok sto sudando sangue @mauroerta 300 negroni se mi spieghi come maledizione fargli computare i tipi delle collezioni referenziate. e' tutto ok a livello di definizioni ecc. ma quando lo uso:

```ts
const article = await cms.data.articles.findOne()
```
`article.author` c'e, ma e' `never`. 

Ho chiesto a ChatGPT in 300 modi e non c'e riuscito, sto avendo il dubbio se questa cosa sia possibile (avendo tutte le collezioni nello stesso oggetto)